components/Card/Card.js

```diff
@@ -1,11 +1,21 @@
 import styled from "styled-components";
 import { IoStar, IoLocationSharp } from "react-icons/io5";
+import FavoriteButton from "../FavoriteButton/FavoriteButton";
+
 import Link from "next/link";
 
-export default function Card({ location }) {
+export default function Card({ location, onToggleFavorite, favorites }) {
+  const isFavorite = favorites.includes(location?.id);
+
```

Comment:
I think it's better to have the logic deciding if this is a favorite or not in the parent component. It's always good to make components "as stupid as possible". You could have a prop `isFavorite` and set this in the parent component when you map your card array.

Reply:
Ralf, thanks for your review.
I agree with you having components as stupid as possible. In this case I don't know the id in the parent component.
That's why I need line 8 in the Card component